lsp-types fails to build, needs package bounds revision

Error:

```
src/Language/LSP/Protocol/Types/Common.hs:171:11: error: [GHC-83865]
    • Expecting one more argument to ‘KeyValue kv’
      Expected a constraint, but ‘KeyValue kv’ has kind ‘* -> Constraint’
    • In the type signature:
        (.=?) :: (KeyValue kv, ToJSON v) => String -> Maybe v -> [kv]
    |
171 | (.=?) :: (J.KeyValue kv, J.ToJSON v) => String -> Maybe v -> [kv]
    |           ^^^^^^^^^^^^^
```

It builds fine with `cabal build --constraint='aeson<2.2.0.0'`

--- a/cabal.project
+++ b/cabal.project
@@ -6,7 +6,7 @@ packages:
 package lsp
   flags: +demo
 
-index-state: 2023-05-18T00:00:00Z
+index-state: 2023-07-09T00:00:00Z
 
 tests: True
 benchmarks: True

--- a/lsp-types/lsp-types.cabal
+++ b/lsp-types/lsp-types.cabal
@@ -68,7 +68,7 @@ library
     UndecidableInstances
 
   build-depends:
-    , aeson             >=1.2.2.0
+    , aeson             >=1.2.2.0 && <2.2
     , base              >=4.11    && <5
     , binary
     , containers

